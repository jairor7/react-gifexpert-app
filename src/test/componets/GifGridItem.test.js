import shallow from "enzyme/shallow";
import GifGridItem from "../../components/GifGridItem";

describe("Test to component GifGridItem", () => {
  const title = "Title";
  const url = "http://localhost/algo.gif";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Should show correct GifGridItem", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Should show correct title", () => {
    const textTitle = wrapper.find("p").at(0).text().trim();
    expect(textTitle).toBe(title);
  });

  test("Should show correct props of the image", () => {
    const img = wrapper.find("img");
    // const propsImg = img.props();  //Para buscar cada pros se puede usar propsImg.src
    const textUrl = img.prop("src");
    const textTitle = img.prop("alt");
    expect(textUrl).toBe(url);
    expect(textTitle).toBe(title);
  });

  test("Img should have class animate__animated", () => {
    const classAnimated = "animate__animated";
    const div = wrapper.find("div");
    const propsDiv = div.prop("className");
    expect(propsDiv.includes(classAnimated)).toBe(true);
  });
});
