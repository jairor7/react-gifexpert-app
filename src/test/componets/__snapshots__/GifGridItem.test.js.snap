// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to component GifGridItem Should show correct GifGridItem 1`] = `
<div
  className="card animate__animated animate__fadeInRight"
>
  <p>
    Title
  </p>
  <img
    alt="Title"
    src="http://localhost/algo.gif"
  />
</div>
`;
